compiler/rustc_session/src/config.rs

```diff
@@ -1236,6 +1236,10 @@ fn default_configuration(sess: &Session) -> Cfg {
         ins_none!(sym::overflow_checks);
     }
 
+    if sess.ub_checks() {
+        ins_none!(sym::ub_checks);
```

Comment:
This is missing it's counter-part below in `CheckCfg::fill_well_known`: https://github.com/rust-lang/rust/blob/6cf853c2ecbd07c939b1386b6803b1d512a8f24b/compiler/rustc_session/src/config.rs#L1392

Something like this should do it:
```rust
        ins!(sym::ub_checks, no_values);
```

Please also follow the guidelines, https://github.com/rust-lang/rust/blob/6cf853c2ecbd07c939b1386b6803b1d512a8f24b/compiler/rustc_session/src/config.rs#L1415-L1426

Reply:
It sounds like you told me that the previous state here is buggy. Why did the tests pass?

Reply:
We don't currently sanity check that `default_configuration` and `CheckCfg::fill_well_known` are synced; we should probably do that, because it's non-trivial.

We rely on them being synced.

We also don't currently automatically enable check-cfg in compiletest, that's why CI doesn't fail. (I will send a PR to enable it by default)

-----

Btw I don't see the cfg added in `tests/ui/check-cfg/well-known-values.rs` nor do I see the documentation being updated at `src/doc/unstable-book/src/compiler-flags/check-cfg.md`.

You will also need to bless the all `tests/ui/check-cfg/` directory.

Reply:
Thanks. All the check-cfg changes look good.

Reply:
Thanks for maintaining check-cfg. It's a great feature.